Decouple TxBuilder from WellKnownKeys

kchulj opened the pull request:
`TxBuilder` could only be used with the well-known keypairs. Now, a transaction can be signed with not-yet-known keypairs.
If one wants to use not-yet-known keypairs, then they can call `signTx`.

Fixes #1348

kchulj commented:
This is an example of a lookup function which can be used with our own `myKeys`. It would be an argument in `signTx`.
```D
public KeyPair lookup (PublicKey pubkey)
{
    KeyPair[] myKeys;
    SecretKey[PublicKey] mapping;
    foreach (const ref key; myKeys)
        mapping[key.address] = key.secret;

    auto result = mapping.byKeyValue().find!(kv => kv.key == pubkey);
    return result.empty ? KeyPair.init : KeyPair(pubkey, result.front);
}
```

AndrejMitrovic commented on source/agora/utils/Test.d:
```diff
@@ -595,6 +598,27 @@ public struct TxBuilder
+
+    ***************************************************************************/
+
+public Transaction signTx (ref TxBuilder txb, TxType type = TxType.Payment,
```
It seems the indentation of this (and the `spendable`) function is wrong. Shouldn't it be indented like the ddoc comment?

AndrejMitrovic replied:
Also I don't see where this function is used?

kchulj replied:
> It seems the indentation of this (and the spendable) function is wrong. Shouldn't it be indented like the ddoc comment?

I simply followed the style of `spendable`, however that's not the case for other ddocs. Perhaps the ddoc for this and `spendable` should have no indentation ? (they are not part of the `TxBuilder` struct)

AndrejMitrovic replied:
Right then the docs should be unindented.

kchulj replied:
> Also I don't see where this function is used?

"One of the goal here is to avoid having to edit every single use of it"
I am thinking of leaving the existing `sign`s, and this one can be used whenever one wants to use a not-well-known KeyPair. Open for suggestions.

Geod24 commented:
`toDelegate` is a nice find. But why do you need the overload then ?

kchulj commented:
> `toDelegate` is a nice find. But why do you need the overload then ?

Yes, the delegate can now be an optional argument.
Thank you @omerfirmak for the help :)

Commit:
Trivial: Fix ddoc indentation

Commit:
Decouple TxBuilder from WellKnownKeys

TxBuilder could only be used with the well-known keypairs.
Now, a transaction can be signed with not-yet-known keypairs.

Fixes #1348

Geod24 reviewed (approved):
LGTM
